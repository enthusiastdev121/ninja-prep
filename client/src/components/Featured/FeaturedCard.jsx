import React from 'react'
import './featured.css'

const FeaturedCard = (props) => {
    return (
        <div className="card ">
            <img className="card-img-top" src={props.discoverCardImgPath} alt="Card cap" />
            <div className="card-body">
                <h2 className="card-title">{props.discoverCardTitle}</h2>
                <p className="card-text subtitle">{props.discoverCardSubtitle}</p>
            </div>
        </div>
    )
}

export default FeaturedCard
export { default as FeaturedCard } from './FeaturedCard'
