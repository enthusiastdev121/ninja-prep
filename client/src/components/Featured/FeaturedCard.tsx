import React from 'react'
import './featured.css'

interface Props {
    discoverCardImgPath: string
    discoverCardTitle: string
}

const FeaturedCard = (props: Props) => {
    return (
        <div className="card mr-5">
            <img className="card-img-top img-fluid" src={props.discoverCardImgPath} alt="Card cap" />
            <div className="card-body">
                <p className="card-title lead text-dark">{props.discoverCardTitle}</p>
            </div>
        </div>
    )
}

export default FeaturedCard
