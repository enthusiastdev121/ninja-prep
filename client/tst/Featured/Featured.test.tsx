import React from 'react'
import FeaturedCard from '../../src/components/Featured/FeaturedCard'
import { shallow } from 'enzyme'

it('renders successfully', () => {
    const wrapper = shallow(<FeaturedCard discoverCardImgPath="" discoverCardTitle="" />)
    expect(wrapper).toBeDefined
})
