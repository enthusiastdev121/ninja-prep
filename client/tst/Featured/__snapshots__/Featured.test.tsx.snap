// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Featured>
    <div
      className="mx-5 px-5 pb-5 pt-5 mb-5"
    >
      <div
        className="mx-5"
      >
        <h2
          className="font-weight-bold pb-4"
        >
          Featured
        </h2>
        <div
          className="card-deck "
        >
          <FeaturedCard
            discoverCardImgPath="./images/discover1.png"
            discoverCardSubtitle="Stacks"
            discoverCardTitle="Stacks 101"
            key="0"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover1.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Stacks 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover2.png"
            discoverCardSubtitle="Queues"
            discoverCardTitle="Queues 101"
            key="1"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover2.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Queues 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover3.png"
            discoverCardSubtitle="Linked List"
            discoverCardTitle="Linked List 101"
            key="2"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover3.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Linked List 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover4.png"
            discoverCardSubtitle="Binary"
            discoverCardTitle="Remembering Binary"
            key="3"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover4.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Remembering Binary
                </p>
              </div>
            </div>
          </FeaturedCard>
        </div>
      </div>
    </div>
  </Featured>
</Router>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Featured>
    <div
      className="mx-5 px-5 pb-5 pt-5 mb-5"
    >
      <div
        className="mx-5"
      >
        <h2
          className="font-weight-bold pb-4"
        >
          Featured
        </h2>
        <div
          className="card-deck "
        >
          <FeaturedCard
            discoverCardImgPath="./images/discover1.png"
            discoverCardSubtitle="Stacks"
            discoverCardTitle="Stacks 101"
            key="0"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover1.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Stacks 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover2.png"
            discoverCardSubtitle="Queues"
            discoverCardTitle="Queues 101"
            key="1"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover2.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Queues 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover3.png"
            discoverCardSubtitle="Linked List"
            discoverCardTitle="Linked List 101"
            key="2"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover3.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Linked List 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover4.png"
            discoverCardSubtitle="Binary"
            discoverCardTitle="Remembering Binary"
            key="3"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover4.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Remembering Binary
                </p>
              </div>
            </div>
          </FeaturedCard>
        </div>
      </div>
    </div>
  </Featured>
</Router>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Featured>
    <div
      className="mx-5 px-5 pb-5 pt-5 mb-5"
    >
      <div
        className="mx-5"
      >
        <h2
          className="font-weight-bold pb-4"
        >
          Featured
        </h2>
        <div
          className="card-deck "
        >
          <FeaturedCard
            discoverCardImgPath="./images/discover1.png"
            discoverCardSubtitle="Stacks"
            discoverCardTitle="Stacks 101"
            key="0"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover1.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Stacks 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover2.png"
            discoverCardSubtitle="Queues"
            discoverCardTitle="Queues 101"
            key="1"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover2.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Queues 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover3.png"
            discoverCardSubtitle="Linked List"
            discoverCardTitle="Linked List 101"
            key="2"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover3.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Linked List 101
                </p>
              </div>
            </div>
          </FeaturedCard>
          <FeaturedCard
            discoverCardImgPath="./images/discover4.png"
            discoverCardSubtitle="Binary"
            discoverCardTitle="Remembering Binary"
            key="3"
          >
            <div
              className="card mr-5"
            >
              <img
                alt="Card cap"
                className="card-img-top img-fluid"
                src="./images/discover4.png"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-title lead text-dark"
                >
                  Remembering Binary
                </p>
              </div>
            </div>
          </FeaturedCard>
        </div>
      </div>
    </div>
  </Featured>
</Router>
`;
